cmd/dataload-init/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	datastorev1alpha1 "github.com/hwameistor/datastore/pkg/apis/client/clientset/versioned/typed/datastore/v1alpha1"
+	datastore "github.com/hwameistor/datastore/pkg/apis/datastore/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"os"
+	"time"
+)
+
+var (
+	nodeName = flag.String("nodename", "", "Node name")
+	subDir   = flag.String("subdir", "", "subdir")
+)
+
+const (
+	NameSpaceEnvVar = "NAMESPACE"
+	pvcNameEnvVar   = "PVC_NAME"
+	apiGroup        = "example.com"
+	version         = "v1alpha1"
+)
+
+func main() {
+	flag.Parse()
+	if *nodeName == "" {
+		log.WithFields(log.Fields{"nodename": *nodeName}).Error("Invalid node name")
+		os.Exit(1)
+	}
+	if *subDir == "" {
+		log.WithFields(log.Fields{"subdir": *subDir}).Error("Invalid subdir path")
+		os.Exit(1)
+	}
+	namespace := os.Getenv(NameSpaceEnvVar)
+	pvcName := os.Getenv(pvcNameEnvVar)
+
+	if namespace == "" || pvcName == "" {
+		log.Fatal("Namespace or PVC Name environment variables are not set.")
+	}
+
+	config, err := getConfig()
+	if err != nil {
+		log.WithError(err).Fatal("Failed to get Kubernetes configuration")
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to create Kubernetes clientset")
+	}
+
+	pvcClient := clientset.CoreV1().PersistentVolumeClaims(namespace)
+	pvc, err := getPersistentVolumeClaim(pvcClient, pvcName)
+	if err != nil {
+		log.WithError(err).Error("Failed to get PVC")
+		return
+	}
+
+	dataSetName := pvc.Spec.VolumeName
+	dataLoadRequest := createDataLoadRequest(dataSetName, *subDir)
+	dsClient, err := datastorev1alpha1.NewForConfig(config)
+	watcher, err := watchCustomResource(dsClient, namespace, dataSetName)
+	if err != nil {
+		log.WithError(err).Error("Failed to start watching custom resource")
+		return
+	}
+	defer watcher.Stop()
+	// 开始计时
+	start := time.Now()
+	if err := createCustomResource(dsClient, dataLoadRequest, namespace); err != nil {
+		log.WithError(err).Error("Failed to create custom resource")
+		return
+	}
+	fmt.Println("Created custom resource")
```

Comment:
debug code ...

Reply:
Thanks, it has been processed. [](https://github.com/hwameistor/datastore/pull/17)